lib/actions/user.js

```diff
@@ -181,3 +188,91 @@ export function deleteUserMonitoredTrip (id) {
     }
   }
 }
+
+/**
+ * Requests a verification code via SMS for the logged-in user.
+ */
+export function requestPhoneVerificationSms (originalUserData, newPhoneNumber) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+
+    if (otpMiddleware) {
+      const { accessToken } = user
+
+      // FIXME: (required by middleware) Only temporarily update the user's phone number
+      //   in the database with the pending one,
+      //   and cache the previous value and verification state so they can be reverted.
+      //   The full user record will be updated upon the user clicking Finish/Save preferences.
+      // TODO: Figure out what should happen if the user refreshes the browser at this stage.
```

Comment:
If there needs to be a change to the middleware design, let's just discuss/handle that now instead of embedding things to fix in this PR.

Reply:
I agree, as noted in the PR description.

Reply:
@binh-dam-ibigroup, can you take the lead on scheduling a call to resolve the backend design so that we can merge/fix this?